blazar/manager/service.py

```diff
@@ -331,7 +332,22 @@ def devices_in_lease(self, lease_id):
         return db_api.devices_in_lease(lease_id)
 
     def list_leases(self, project_id=None, query=None):
-        return db_api.lease_list(project_id)
+        limit = query.get("limit")
+        if strutils.is_int_like(limit):
+            limit = int(limit)
+        elif limit is not None:
+            raise common_ex.InvalidInput(
+                f'limit must be integer, got {limit}')
```

Comment:
is there somewhere else that input validation happens? mostly I'm worried that some garbage could be passed as limit, then get converted to a ridiculous int

also, what happens if limit=none? I see it gets passed to sqlalchemyutils.paginate_query, does that revert to the previous case of no pagination?